src/ranged.cpp

```diff
@@ -1862,13 +1862,16 @@ item::sound_data item::gun_noise( bool const burst ) const
 
     noise = std::max( noise, 0 );
 
+    // Grenade launchers
     if( ammo_type() == ammotype( "40mm" ) ) {
         return { 8, _( "Thunk!" ) };
 
-    } else if( typeId() == "hk_g80") {
+    // Railguns
+    } else if( ammotype() == "12mm" || ammotype() == "metal_rail" ) {
```

Comment:
Gives a [kerblam](http://i.imgur.com/NRWOPCb.png) now. This if will never evaluate to true, `ammo_type()` is the function you're looking for and [its type is ammotype](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/item.cpp#L4255), can't compare with a string. This:
`} else if( ammo_type() == ammotype( "12mm" ) || ammo_type() == ammotype( "metal_rail" ) ) {`
gives the [cracklies](http://i.imgur.com/sPP9fYJ.png).